data_structures/src/Bag/Bag.cs

```diff
@@ -6,41 +6,80 @@ namespace ADT
 {
     public class Bag<T>: IBag<T>
     {
-        public Bag(int size = 10) { }
+        private T[] _bag { get; set; }
+        private int _fullSpots { get; set; }
+
+        public Bag(int size = 10) { 
+            _bag = new T[size];
+        }
 
         public void Add(T item)
         {
-            throw new NotImplementedException();
+            try
+            {
+                if(_fullSpots == 0)
+                {
+                    _bag[_fullSpots] = item;   
+                }else
+                {
+                    _bag[_fullSpots] = item;
+                }
```

Comment:
The conditional results in the same action being performed. Best to check if the number of items is greater than or equal to the max size and thrown an exception if true.